src/libraries/System.Text.Json/tests/Serialization/ExtensionDataTests.cs

```diff
@@ -903,8 +905,6 @@ public static void DeserializeIntoImmutableDictionaryProperty()
             JsonSerializer.Deserialize<ClassWithExtensionPropertyPrivateConstructor>(@"{}");
             JsonSerializer.Deserialize<ClassWithExtensionPropertyPrivateConstructorJsonElement>(@"{}");
 
-            Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<ClassWithExtensionPropertyAsImmutable>("{\"hello\":\"world\"}"));
```

Comment:
Please add JSON properties to the above tests and verify the results of the deserialization operations.

Reply:
I think the tests above are to verify that no exceptions are thrown when the extension data isn't used but does when it tries to deserialize into it. I removed these two since they're no longer accurate and it does serialize